### PR TITLE
Add a compiler configuration for Intel 19 on sems-rhel7 for compatibility testing

### PR DESCRIPTION
**Pull request by PhilMiller**
@trilinos/

## Motivation

Versions of the Intel compiler below 19 don't provide C++17 support. This will be required by Kokkos in release 3.7, and on its `develop` branch soon after 3.6. Testing the newer compilers before the later release is proposed for integration in Trilinos will allow any bugs or necessary workarounds to be identified sooner.

## Testing

Tested with ctest in EMPIRE's configuration of selected options and packages, and with EMPIRE's own suite.

### CODE REVIEW

**Comment by PhilMiller** [comment]
@rppawlo

**Review by e10harvey — COMMENTED** [review]
Thanks, @PhilMiller ! This looks fine but is missing the driver and supported build changes.

I've opened a PR with those changes although I am not sure what build options you're looking to support. I started testing via:
```
[Trilinos (pm/intel19)]$ gst -uno
# On branch pm/intel19
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#       new file:   cmake/ctest/drivers/atdm/sems-rhel7/drivers/Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug.sh
#       new file:   cmake/ctest/drivers/atdm/sems-rhel7/drivers/Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug.sh
#       modified:   cmake/std/atdm/sems-rhel7/all_supported_builds.sh
#
# Untracked files not listed (use -u option to show untracked files)
[Trilinos (pm/intel19)]$ git commit -m 'atdm/sems-rhel7: Add drivers and update supported builds'
[pm/intel19 f3c51aa] atdm/sems-rhel7: Add drivers and update supported builds
 3 files changed, 12 insertions(+)
 create mode 100755 cmake/ctest/drivers/atdm/sems-rhel7/drivers/Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug.sh
 create mode 100755 cmake/ctest/drivers/atdm/sems-rhel7/drivers/Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug.sh
[Trilinos (pm/intel19)]$ cd ../
[TRILINOS.base]$ mkdir Trilinos-pr-10047
[TRILINOS.base]$ cd Trilinos-pr-10047/
[Trilinos-pr-10047]$ ln -s ../Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
[Trilinos-pr-10047]$ ls
ctest-s-local-test-driver.sh
[Trilinos-pr-10047]$ nohup ./ctest-s-local-test-driver.sh sems-rhel7-intel-19.0.5-openmp-shared-release-debug sems-rhel7-intel-19.0.5-openmp-shared-debug &> ctest-s-local-test-dr\
iver.out &
[Trilinos-pr-10047]$ tailf ctest-s-local-test-driver.out

Thu Dec 23 07:19:41 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug/smart-jenkins-driver.out
```
which should post [here on CDash](https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2021-12-23&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=sems-rhel7&field2=buildname&compare2=63&value2=intel-19&field3=groupname&compare3=63&value3=Experimental).

You may have to address failures depending on what the testing results show.

**Comment by PhilMiller** [comment]
@e10harvey Thanks for filling in the missing bits I was unaware of.

My goal is roughly to cover the same usage of the Intel compiler as EMPIRE has for Intel 18 right now - "intel-opt-openmp-static". We've already identified one set of cases where building with Intel without optimization numerically perturbs some unstable tests, and concluded that they could be disabled.

I just don't want to let the day of the Kokkos 3.7 release and merge into Trilinos, when Intel 19+ will have to replace Intel 17/18 turn into a major fire drill for downstream applications teams

**Comment by PhilMiller** [comment]
Oh, and to be clear, the very few tests that got disabled were the only problem that came up. If we can get this integrated in the pipelines, then we can also test regularly with it and avoid any unexpected compiler ICEs

**Comment by PhilMiller** [comment]
The test failures seen on CDash are seemingly all of the form
```

/home/eharvey/dev/TRILINOS.base/Trilinos-pr-10047/Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug/SRC_AND_BUILD/BUILD/packages/panzer/adapters-stk/example/PoissonExample/PanzerAdaptersSTK_PoissonExample.exe: symbol lookup error: /projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/lib/libmca_common_verbs.so.7: undefined symbol: ompi_common_verbs_usnic_register_fake_drivers
```

So, it would seem we have some linking issue to address

**Comment by rppawlo** [comment]
@e10harvey or @bartlettroscoe - could one of you review this?

File: cmake/std/atdm/cts1empire/environment_new.sh [hunk 1]
```diff
@@ -55,8 +62,8 @@ fi
 export OMPI_CXX=`which icpc`
 export OMPI_CC=`which icc`
 export OMPI_FC=`which ifort`
-export ATDM_CONFIG_LAPACK_LIBS="-mkl"
-export ATDM_CONFIG_BLAS_LIBS="-mkl"
+export ATDM_CONFIG_LAPACK_LIBS="-mkl -L${MKL_LIB}"
```
Comment (bartlettroscoe) [inline cmake/std/atdm/cts1empire/environment_new.sh:65]: Where is `MKL_LIB` coming from?

File: cmake/std/atdm/sems-rhel7/environment.sh [hunk 2]
```diff
@@ -46,14 +46,16 @@ elif [[ "$ATDM_CONFIG_COMPILER" == "INTEL"* ]]; then
   if [[ "$ATDM_CONFIG_COMPILER" == "INTEL" ]] ; then
     export ATDM_CONFIG_COMPILER=INTEL-18.0.5
```
Comment (bartlettroscoe) [inline cmake/std/atdm/sems-rhel7/environment.sh:47]: Should the default intel version not be updated to intel-19.0.5 as well?  I think EMPIRE is the only real customer for this and I think only EMPIRE developers have been leaving the version number off of 'intel' in the build name when they set up a configuration.  I think other users tend to supply the exact Intel version.

File: packages/panzer/disc-fe/test/evaluator_tests/normals.cpp [hunk 3]
```diff
@@ -228,7 +228,8 @@ typedef Traits::Residual ResidualType;
 typedef Traits::Jacobian JacobianType;
 
 UNIT_TEST_GROUP(ResidualType)
-UNIT_TEST_GROUP(JacobianType)
+// Commented out to avoid ICE in Intel 2021.3.0
+//UNIT_TEST_GROUP(JacobianType)
```
Comment (bartlettroscoe) [inline packages/panzer/disc-fe/test/evaluator_tests/normals.cpp:232]: It is surprising that these tests need to be disabled for this Intel 19.0.5 build.  Are there not other nightly Intel 19 builds running already that run these Panzer tests?

**Review by bartlettroscoe — COMMENTED** [review]
These changes seem pretty straightforward to me.  I did not check out the branch and manually test it but the changes seem reasonable.  (Someone else should do that carefully.)  I just have a few comments/questions below.

Note that I did not check the "Approve" button with this PR review because I don't think I have the authority to approve changes to the ATDM Trilinos configuration system anymore and allow merges to 'develop'.

File: cmake/std/atdm/cts1empire/environment_new.sh (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/std/atdm/cts1empire/environment_new.sh:65]: I think `-mkl` is a special compiler/linker flag that is directly understood by the Intel compiler and it does not need to know a library path with `-L` to find the MKL libraries associated with the compiler. Perhaps `-mkl` just instructs the Intel compiler to put the right `-l<lib>` arguments on the link line but you can change what MKL implementation is used by setting `-L<mkl-lib-dir>`?  Is that what the SPARC modules are doing here on CTS-1?

**Comment by bartlettroscoe** [comment]
@rppawlo and @PhilMiller, I thought that EMPIRE could not build with Intel-19 due to a bug in Intel-19?  Did something change?

**Comment by e10harvey** [comment]
@PhilMiller: Would you please post the cdash link to the test results for these change?

**Comment by PhilMiller** [comment]
@bartlettroscoe Sorry for the confusion in some of the changes. I pushed additional changes to this branch related to testing Intel 2021.3.0 compilers. I'll force-push this back, and push a separate branch for those.

File: cmake/std/atdm/sems-rhel7/environment.sh (hunk 2 above)
Comment (PhilMiller) [reply cmake/std/atdm/sems-rhel7/environment.sh:47]: I'd be thrilled to make that switch, if that's what people want.

File: cmake/std/atdm/cts1empire/environment_new.sh (hunk 1 above)
Comment (PhilMiller) [reply cmake/std/atdm/cts1empire/environment_new.sh]: That was a further hack for a test build I've done with Intel 2021.3.0 and the corresponding MKL.

The `-mkl` flag (deprecated, per Intel, in favor of `-qmkl`) is supposed to add the appropriate includes, library paths, and linked libraries to the command lines it generates, but it seems not to in the one environment I have available to test it on, SNL's Eclipse cluster. I got linking errors for it not being able to find the MKL libraries at link time.

At any rate, that's irrelevant to this PR, which I've reduced back to just the Intel 19 changes

File: packages/panzer/disc-fe/test/evaluator_tests/normals.cpp (hunk 3 above)
Comment (PhilMiller) [reply packages/panzer/disc-fe/test/evaluator_tests/normals.cpp]: Again, this was for Intel 2021.3.0. It's fine on 19.0.5

**Comment by PhilMiller** [comment]
> @rppawlo and @PhilMiller, I thought that EMPIRE could not build with Intel-19 due to a bug in Intel-19? Did something change?

That might have been the case a few years ago with 19.0.1 or something, but 19.0.5 builds and passes the full suite of tests

File: packages/panzer/disc-fe/test/evaluator_tests/normals.cpp (hunk 3 above)
Comment (PhilMiller) [reply packages/panzer/disc-fe/test/evaluator_tests/normals.cpp]: That failure is tracked here: https://github.com/trilinos/Trilinos/issues/10097

File: cmake/std/atdm/sems-rhel7/environment.sh (hunk 2 above)
Comment (PhilMiller) [reply cmake/std/atdm/sems-rhel7/environment.sh:47]:
```suggestion
    export ATDM_CONFIG_COMPILER=INTEL-19.0.5
```

Comment (PhilMiller) [reply cmake/std/atdm/sems-rhel7/environment.sh:47]: Maybe we should wait to make that change until after the build is available, and has been run through the more extensive set of tests that EMPIRE runs less often than on every change

**Comment by PhilMiller** [comment]
There's something missing in the overall script modifications, because I tried to run the command to build, test, and submit results to CDash, but I found the 2018 compilers running, and so dug in to see this:

```
$ grep 18.0.5 Trilinos-atdm-sems-rhel7-intel-19.0.5-openmp-shared-release-debug/SRC_AND_BUILD/BUILD/CMakeCache.txt  | grep bin
CMAKE_CXX_COMPILER:STRING=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpicxx
CMAKE_C_COMPILER:STRING=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpicc
CMAKE_Fortran_COMPILER:FILEPATH=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpif90
MPI_EXEC:FILEPATH=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpiexec
NC_CONFIG:FILEPATH=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/bin/nc-config
NETCDF_NCDUMP_BINARY:FILEPATH=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/bin/ncdump
XIAR:FILEPATH=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/bin/xiar
```

**Comment by e10harvey** [comment]
> There's something missing in the overall script modifications, because I tried to run the command to build, test, and submit results to CDash, but I found the 2018 compilers running, and so dug in to see this:

Can you run the `load-env.sh` script for this environment and see if `env | grep 18.0.5` returns anything?

**Comment by PhilMiller** [comment]
```
[pbmille@ascic169 Trilinos]$ git show
commit bfbd3399458d78aa7c79db2edb1529f30d152dd1 (HEAD)
Merge: f7fdd76 f3c51aa
Author: Phil Miller <phil.miller@intensecomputing.com>
Date:   Thu Dec 23 09:30:02 2021 -0800

    Merge pull request #1 from e10harvey/pm/intel19
    
    atdm/sems-rhel7: Add drivers and update supported builds


[pbmille@ascic169 Trilinos]$ source cmake/std/atdm/load-env.sh atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug
Hostname 'ascic169' matches known ATDM host 'ascic169' and system 'sems-rhel7'
Setting compiler and build options for build-name 'atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug'
Using SEMS RHEL7 compiler stack INTEL-18.0.5 to build DEBUG code with Kokkos node type OPENMP
	WARNING: sems-archive-gcc dependency already found but does not match listed dependency sems-archive-gcc/6.1.0
	I will use the sems-archive-gcc you have loaded but correct behavior is not guaranteed
atdm-mkl/18.0.5(115):ERROR:105: Unable to locate a modulefile for 'sems-gcc/4.9.3'


[pbmille@ascic169 Trilinos]$ module list
Currently Loaded Modulefiles:
  1) sems-archive-env                     3) sems-archive-cmake/3.19.1            5) sems-archive-gcc/7.2.0               7) atdm-env                             9) sems-archive-openmpi/1.10.1         11) sems-archive-netcdf/4.7.3/parallel  13) sems-archive-boost/1.59.0/base
  2) sems-archive-git/2.10.1              4) sems-archive-ninja_fortran/1.10.0    6) sems-archive-intel/18.0.5            8) atdm-mkl/18.0.5                     10) sems-archive-hdf5/1.10.6/parallel   12) sems-archive-zlib/1.2.8/base        14) sems-archive-superlu/4.3/base


[pbmille@ascic169 Trilinos]$ env | grep 18.0.5
SEMS_INTEL_ROOT=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base
MANPATH=/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/share/man:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/4.8.5/base/share/man:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/share/man:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/4.8.5/openmpi/1.10.1/parallel/share/man:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/share/man:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/4.8.5/openmpi/1.10.1/share/man:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/share/man:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/7.2.0/base/share/man:/projects/sems/install/rhel7-x86_64/sems/utility/ninja_fortran/1.10.0/share/man:/projects/sems/install/rhel7-x86_64/sems/utility/ninja_fortran/1.10.0/man:/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.19.1/share/man:/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.19.1/man:/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/share/man:/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/man:/ascldap/users/pbmille/repos/git-subrepo/man:/opt/puppetlabs/puppet/share/man
SEMS_ZLIB_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base
SEMS_NETCDF_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/lib
INTEL_LICENSE_FILE=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/Licenses/intel-Linux-SRN.lic
SEMS_SUPERLU_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base/include
SEMS_OPENMPI_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/include
MPICC=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpicc
LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/4.8.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/4.8.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/lib:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/4.8.5/openmpi/1.10.1/lib
SEMS_NETCDF_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/include
OMPI_FC=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/bin/ifort
PNETCDF_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel
SEMS_SUPERLU_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base
SEMS_OPENMPI_ROOT=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1
SEMS_ZLIB_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/include
LD_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/4.8.5/base/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/4.8.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/4.8.5/openmpi/1.10.1/parallel/lib:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/lib:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/4.8.5/openmpi/1.10.1/lib:/projects/sems/install/rhel7-x86_64/atdm/utility/mkl/18.0.5/compilers_and_libraries_2018/linux/mkl/lib/intel64/:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/mkl/lib/intel64:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/lib/intel64:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/7.2.0/base/lib64:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/7.2.0/base/lib:/projects/sems/install/rhel7-x86_64/sems/utility/ninja_fortran/1.10.0/lib:/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.19.1/lib:/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/lib
SEMS_MKL_VERSION=18.0.5
MKL_LIB=/projects/sems/install/rhel7-x86_64/atdm/utility/mkl/18.0.5/compilers_and_libraries_2018/linux/mkl/lib/intel64/
SEMS_HDF5_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/include
ATDM_CONFIG_COMPILER=INTEL-18.0.5
SEMS_ZLIB_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/lib
CPATH=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/gcc/4.8.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/gcc/4.8.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/4.8.5/base/include:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/include:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/4.8.5/openmpi/1.10.1/parallel/include:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/include:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/4.8.5/openmpi/1.10.1/parallel/include:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/include:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/4.8.5/openmpi/1.10.1/include
SEMS_MKL_ROOT=/projects/sems/install/rhel7-x86_64/atdm/utility/mkl/18.0.5/compilers_and_libraries_2018/linux/mkl
SEMS_BOOST_INCLUDE_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base/include
SEMS_COMPILER_VERSION=18.0.5
NETCDF_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel
PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/bin:/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/4.8.5/openmpi/1.10.1/parallel/bin:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/bin:/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/4.8.5/openmpi/1.10.1/parallel/bin:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/4.8.5/openmpi/1.10.1/bin:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/mkl/bin:/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/bin:/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/7.2.0/base/bin:/projects/sems/install/rhel7-x86_64/sems/utility/ninja_fortran/1.10.0/sbin:/projects/sems/install/rhel7-x86_64/sems/utility/ninja_fortran/1.10.0/bin:/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.19.1/sbin:/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.19.1/bin:/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/sbin:/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/bin:/ascldap/users/pbmille/repos/git-subrepo/lib:/usr/local/anaconda/3-2020-02/bin/:/usr/lib64/qt-3.3/bin:/ascldap/users/pbmille/perl5/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin
SEMS_BOOST_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base/lib
ATDM_CONFIG_HDF5_LIBS=/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib/libhdf5_hl.so;/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib/libhdf5.so;/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/lib/libz.so;-ldl
SEMS_SUPERLU_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/superlu/4.3/intel/18.0.5/base/lib
SEMS_OPENMPI_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/lib
_LMFILES_=/projects/sems/modulefiles/projects/sems-archive-env:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/utility/sems-archive-git/2.10.1:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/utility/sems-archive-cmake/3.19.1:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/utility/sems-archive-ninja_fortran/1.10.0:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/compiler/sems-archive-gcc/7.2.0:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/compiler/sems-archive-intel/18.0.5:/projects/sems/modulefiles/projects/atdm-env:/projects/sems/modulefiles/rhel7-x86_64/atdm/utility/atdm-mkl/18.0.5:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/compiler/sems-archive-openmpi/1.10.1:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/tpl/sems-archive-hdf5/1.10.6/parallel:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/tpl/sems-archive-netcdf/4.7.3/parallel:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/tpl/sems-archive-zlib/1.2.8/base:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/tpl/sems-archive-boost/1.59.0/base:/projects/sems/modulefiles/rhel7-x86_64/sems-archive/tpl/sems-archive-superlu/4.3/base
HDF5_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel
SEMS_COMPILER_ROOT=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base
SEMS_NETCDF_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel
LOADEDMODULES=sems-archive-env:sems-archive-git/2.10.1:sems-archive-cmake/3.19.1:sems-archive-ninja_fortran/1.10.0:sems-archive-gcc/7.2.0:sems-archive-intel/18.0.5:atdm-env:atdm-mkl/18.0.5:sems-archive-openmpi/1.10.1:sems-archive-hdf5/1.10.6/parallel:sems-archive-netcdf/4.7.3/parallel:sems-archive-zlib/1.2.8/base:sems-archive-boost/1.59.0/base:sems-archive-superlu/4.3/base
MKL_HOME=/projects/sems/install/rhel7-x86_64/atdm/utility/mkl/18.0.5/compilers_and_libraries_2018/linux/mkl
MKL_INCLUDE=/projects/sems/install/rhel7-x86_64/atdm/utility/mkl/18.0.5/compilers_and_libraries_2018/linux/mkl/include/
OMPI_CC=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/bin/icc
ATDM_CONFIG_NETCDF_LIBS=/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/lib/libnetcdf.so;/projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/intel/18.0.5/openmpi/1.10.1/parallel/lib/libpnetcdf.a;/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib/libhdf5_hl.so;/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib/libhdf5.so;/projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/intel/18.0.5/base/lib/libz.so;-ldl;-lcurl
BOOST_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base
SEMS_HDF5_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel
OMPI_CXX=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/base/bin/icpc
SEMS_HDF5_LIBRARY_PATH=/projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/intel/18.0.5/openmpi/1.10.1/parallel/lib
SEMS_BOOST_ROOT=/projects/sems/install/rhel7-x86_64/sems/tpl/boost/1.59.0/intel/18.0.5/base
SEMS_MPI_ROOT=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1
MPIF90=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpif90
MPICXX=/projects/sems/install/rhel7-x86_64/sems/compiler/intel/18.0.5/openmpi/1.10.1/bin/mpicxx


[pbmille@ascic169 Trilinos]$ env | grep 19.0.5
ATDM_CONFIG_JOB_NAME=atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug
ATDM_CONFIG_BUILD_NAME=atdm-sems-rhel7-intel-19.0.5-openmp-shared-debug
```

**Review by e10harvey — APPROVED** [review]

**Comment by PhilMiller** [comment]
In the full CDash submission based on the commands @e10harvey posted, 
https://testing.sandia.gov/cdash/index.php?project=Trilinos&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=intel-19.0.5&field2=buildstarttime&compare2=84&value2=NOW&field3=site&compare3=61&value3=ascic169

There's one common build error that's repeated across many executables:
```
ld: packages/intrepid2/unit-test/Discretization/Basis/HGRAD_QUAD_C1_FEM/OpenMP/CMakeFiles/Intrepid2_unit-test_Discretization_Basis_HGRAD_QUAD_C1_FEM_OpenMP_Test_01.dir/test_01.cpp.o(.text._ZNK9Intrepid25BasisIN6Kokkos6OpenMPEddE9getDofTagEi[_ZNK9Intrepid25BasisIN6Kokkos6OpenMPEddE9getDofTagEi]+0x45): unresolvable R_X86_64_TPOFF32 relocation against symbol `_ZN6Kokkos4Impl22SharedAllocationRecordIvvE18t_tracking_enabledE'
ld: final link failed: Nonrepresentable section on output
```

**Comment by PhilMiller** [comment]
From what I can find online, there seems to be a mismatch among objects built with or without `-fPIC`. They should all be getting built with `-fPIC` in this configuration.

**Comment by e10harvey** [comment]
I am merging this because:
1. @ndellingwood found that the ld errors are related to kokkos and he will be filing an issue in Kokkos to get this resolved.
2. None of these changes will break existing 'official' atdm builds.

**Comment by ndellingwood** [comment]
Link to the kokkos issue: https://github.com/kokkos/kokkos/issues/4722

**Pull request merged**